src/belt_manager.py

```diff
@@ -0,0 +1,205 @@
+        if toggleBelt:
+            wait(0.2,0.3)
+            keyboard.send(self._config.char["show_belt"]) #toggle belt off
+            wait(0.2,0.3)
+        #Logger.debug(f"belt_contents:{belt_contents}, belt_height:{belt_contents[:,keepRows].shape[1]}")
+        return belt_contents, belt_contents[:,keepRows].shape[1]
+
+    def drop_wrong_belt_pots(self, belt_contents: np.ndarray, toggleBelt: bool):
+        #iterate through every belt position from top to bottom
+        if toggleBelt:
+            wait(0.2,0.3)
+            keyboard.send(self._config.char["show_belt"]) #toggle belt on
+            wait(0.2,0.3)
+        #Logger.debug(f"{belt_slots}")
+        for y in range(belt_contents.shape[1]):
+            y_center=int(round(self._config.ui_pos["potion1_y"] + y*self._config.ui_pos["potion_height"] + self._config.ui_pos["potion_height"]/2,1))
+            for x in range(4):
+                x_center=int(round(self._config.ui_pos["potion1_x"] + x*self._config.ui_pos["potion_width"] + self._config.ui_pos["potion_width"]/2,1))
+                detected=belt_contents[x][y]
+                expected=self._config.char['belt_slots'][x]
+                # if detected potion type not empty, no_match, or expected then drop unless detected type is "full_rejuvenation_potion" AND expected is "rejuvenation_potion"
+                if (detected not in ("empty", "no_match", expected)) and (not (detected == "full_rejuvenation_potion" and expected == "rejuvenation_potion")):
+                    Logger.debug(f"Belt slot({x},{y}) detected {detected} but expected {expected}, drop")
+                    mouse.move(x_center, y_center, randomize=[2, 2], delay_factor=[1, 1.5])
+                    wait(0.2, 0.3)
+                    mouse.click(button="left")
+                    wait(0.2, 0.3)
+                    mouse.move(1920/2*self._config.res["scale"], 1080/2*self._config.res["scale"], randomize=[5, 5], delay_factor=[1,1.5]) #NEED TO FIX THIS LINE SO NOT HARD CODED
+                    wait(0.2, 0.3)
+                    mouse.click(button="left")
+                    wait(0.2, 0.3)
+                # if potion is equal to expected or detected type is "full_rejuvenation_potion" AND expected is "rejuvenation_potion"
+                elif (detected == expected) or (detected == "full_rejuvenation_potion" and expected == "rejuvenation_potion"):
+                    #add to _potions_remaining count
+                    self.potions_remaining[x] = self.potions_remaining[x] +1
+
+        if toggleBelt:
+            wait(0.2,0.3)
+            keyboard.send(self._config.char["show_belt"]) #toggle belt off
+            wait(0.2,0.3)
+
+                #if belt_contents[x][y] is not self._config.char
+
+    def fill_up_belt_from_inventory(self):
+        """
+        Fill up your belt with pots from the inventory e.g. after death. It will open and close invetory by itself!
+        Updated 11/2021 to respect assigned potion columns
```

Comment:
No need to put any dates or anything about changes. That's why we have source control, and commit messages : )

Reply:
Thank you for your comments! Will close the PR for now based on our discord discussion.